Objects/Objects/Geometry/Mesh.cs

```diff
@@ -38,14 +40,115 @@ public Mesh()
 
     }
 
-    public Mesh(double[] vertices, int[] faces, int[] colors = null, double[] texture_coords = null, string units = Units.Meters, string applicationId = null)
+    public Mesh(List<double> vertices, List<int> faces, List<int> colors = null, List<double> texture_coords = null, string units = Units.Meters, string applicationId = null)
     {
-      this.vertices = vertices.ToList();
-      this.faces = faces.ToList();
-      this.colors = colors?.ToList();
-      this.textureCoordinates = texture_coords?.ToList();
+      this.vertices = vertices;
+      this.faces = faces;
+      this.colors = colors ?? this.colors;
+      this.textureCoordinates = texture_coords?? this.textureCoordinates;
       this.applicationId = applicationId;
       this.units = units;
     }
+    
+    public Mesh(double[] vertices, int[] faces, int[] colors = null, double[] texture_coords = null, string units = Units.Meters, string applicationId = null)
+    : this(
+    vertices.ToList(),
+      faces.ToList(),
+      colors?.ToList(),
+      texture_coords?.ToList(),
+      applicationId,
+      units
+    )
+    { }
+
+    #region Convenience Methods
+    
+    public int VerticesCount => vertices.Count / 3;
+    public int TextureCoordinatesCount => textureCoordinates.Count / 2;
+
```

Comment:
Add `GetPoints()` convenience method as in `Curve` class: https://github.com/specklesystems/speckle-sharp/blob/7fba43413cc7e2edba2d88779d94901ffb838cfa/Objects/Objects/Geometry/Curve.cs#L63